portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,32 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/form-handler")
+public class FormHandlerServlet extends HttpServlet {
+
+  // method that redirects user to Neal's Linkedin or Github depending
+  // on form input
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    boolean visitLinkedin = Boolean.parseBoolean(getParameter(request, "linkedin-btn", "false"));
+    if(visitLinkedin) {
+        response.sendRedirect("https://www.linkedin.com/in/nealdavar/");
```

Comment:
For extra credit: why would one prefer this mechanism vs doing it client side?

Reply:
Since servers are typically going to be more powerful than clients' browsers, dynamic rendering on a webpage will be quicker with a server-side approach like this as opposed to a client-side approach.

Reply:
The performance tradeoffs of sever-side rendering are interesting but that's *not* what you're doing here.

Reply:
Ah okay. Could I ask what I am doing here / what you were looking for?

Reply:
The server is doing the redirection. This costs latency but lets you the server author know that the user has done it.